File: GitLab/21_hi023/APP/CleanerTest2/app/src/main/java/com/example/cleanertest2/MainActivity.java
package com.example.cleanertest2;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.widget.Button;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class MainActivity extends AppCompatActivity {

    Handler handler = new Handler();
    testData td = new testData();
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button = findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                final String urlStr = "https://www.naver.com/";

                new Thread(new Runnable() {
                    @Override
                    public void run() {
                        request(urlStr);
                    }
                }).start();

                Intent intent = new Intent(getApplicationContext(), SensorActivity.class);
                td.setnumber(3);
                intent.putExtra("data", td);
                startActivity(intent);
            }
        });
    }

    public void request(String urlStr) {
        String str="";
        StringBuilder output = new StringBuilder();
        try {
            URL url = new URL(urlStr);

            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
            if (conn != null) {
                conn.setConnectTimeout(10000);
                conn.setRequestMethod("GET");
                conn.setDoInput(true);

                int resCode = conn.getResponseCode();
                BufferedReader reader = new BufferedReader(new InputStreamReader(conn.getInputStream()));
                String line = null;
                while (true) {
                    line = reader.readLine();
                    if (line == null) {
                        break;
                    }
                    str=str+line;
                    //output.append(line + "\n");
                }
                reader.close();
                conn.disconnect();
            }
        } catch (Exception ex) {
            //println("예외 발생함 : " + ex.toString());
        }
        td.setstring(str);
        //println("응답 -> " + output.toString());
    }
}

File: app/testapp_7_18/app/src/main/java/com/example/testapp_7_18/MainActivity.kt
package com.example.testapp_7_18

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
    fun clickSensor(view: View){
        Toast.makeText(view.context,  "온습도 센서를 서버에서 받아옵니다.", Toast.LENGTH_LONG).show()
        var intent = Intent(this, SensorActivity::class.java)
        startActivity(intent)
    }
    fun clickStart(view: View){
        var intent = Intent(this, StartActivity::class.java)
        startActivity(intent)
    }
    fun clickDocument(view: View){
        var intent = Intent(this, DocumentActivity::class.java)
        startActivity(intent)
    }
}